syntax/src/lib.rs

```diff
@@ -0,0 +1,31 @@
+#![no_std]
+
+#[macro_use]
+extern crate alloc;
+
+#[cfg(any(test, feature = "std"))]
+extern crate std;
+
+pub use miden_core::{Felt, FieldElement, StarkField, Word, prettier, utils::DisplayHex};
+
+pub mod ast;
+pub mod diagnostics;
+pub mod library;
+mod parse;
+pub mod parser;
+mod sema;
+pub mod testing;
+
+pub use self::{
+    diagnostics::{
+        DefaultSourceManager, Report, SourceFile, SourceId, SourceManager, SourceSpan, Span,
+        Spanned,
+    },
+    library::{
+        KernelLibrary, Library, LibraryError, LibraryNamespace, LibraryPath, LibraryPathComponent,
+        PathError, Version, VersionError,
+    },
+    parse::{Parse, ParseOptions},
+    parser::{ModuleParser, ParsingError},
+    sema::SemanticAnalysisError,
+};
```

Comment:
nit: probably not for this PR but there is some weirdness in how docs get built for this crate. For example, things like `Library`, `ModuleParser` etc. appear as re-export. This is how it looks like for me:

<img width="1019" alt="Screenshot 2025-06-27 at 4 34 01 PM" src="https://github.com/user-attachments/assets/8dcf0737-19f4-4b57-a5f1-f925af50a787" />

Reply:
Makes sense. I was just surprised as docs of our other crates (e.g., `miden-assembly`, `miden-mast-package`, `miden-vm`) don't start out with a long list of re-exports from self. We haven't paid too much attention to our Rust docs - so, fine to leave as is for now, but also if there is an alternative way to do things to make the docs here look more like other crate docs, that may be beneficial even at this stage.

Reply:
It appears the "re-exports" section is displayed here because the modules they are exported from are public, so the they aren't "inlined" as described by the rustdoc documentation. This doesn't happen with `miden-assembly` for example, because the modules it re-exports from are not public.

I've marked the re-exports with `#[doc(hidden)]` to suppress them being displayed in the generated docs under the re-exports section, except those being exported from private modules. This will align with what you were expecting to see.